### ISSUE TITLE
ERROR: Cannot open file Input/carbon.in: No such file or directory

### ISSUE THREAD

**Issue by dschlaep** [Burke-Lauenroth-Lab/rSOILWAT2#74]
- `devtools::run_examples()`: all example for `sw_exec` with input data from files fail with error
> ERROR: Cannot open file Input/carbon.in: No such file or directory

This is despite that the file `Input/carbon.in` is present:
```
list.files(path_demo, pattern = "carbon", recursive = TRUE)
```
> [1] "Input/carbon.in"

The same error is reproduced by running any of the examples by hand:
```
path_demo <- system.file("extdata", "example1", package = "rSOILWAT2")
sw_in2 <- sw_inputDataFromFiles(dir = path_demo, files.in = "files_v31.in")
sw_out2 <- sw_exec(inputData = sw_in2)
```

> Exit.. ERROR: Cannot open file Input/carbon.in: No such file or directory
> Error in sw_exec(inputData = sw_in2) : @ generic.c LogError

- `devtools::test()` unit tests fail with same error

**Comment by Zachary-Kramer** [comment]
Since adding the CSV lookup, rSOILWAT2 does not open carbon.in, only SOILWAT2 standalone does. ppm data is communicated by swCarbon. So, without taking a look at this right now, I'm assuming something is out of date. I can take a look later.

**Comment by dschlaep** [comment]
As above
```
path_demo <- system.file("extdata", "example1", package = "rSOILWAT2")
sw_in2 <- sw_inputDataFromFiles(dir = path_demo, files.in = "files_v31.in")
sw_out2 <- sw_exec(inputData = sw_in2)
```
All the data should be contained in variable `sw_in2` including the CO2 data -- all data should be read in from files by the call `sw_inputDataFromFiles(dir = path_demo, files.in = "files_v31.in")` -- this does not fail.

No code, that is executed as a result of the call `sw_exec(inputData = sw_in2)`, should attempt to read the file `Input/carbon.in` -- yet, something behaves otherwise and fails.

**Comment by Zachary-Kramer** [comment]
> No code, that is executed as a result of the call sw_exec(inputData = sw_in2), should attempt to read the file Input/carbon.in -- yet, something behaves otherwise and fails.

There is only one culprit I can think of. There is one function that behaves differently depending on whether or not `RSOILWAT` is defined. If it is defined, it uses swCarbon. If it is not defined, it uses carbon.in. Is there a possibility that these tests are not defining `RSOILWAT`?

I can test this tomorrow, but if you know off of the top of your head then that'd be quicker

**Comment by dschlaep** [comment]
`RSOILWAT` is defined when the package is compiled and installed.

I experience the above errors during run-time as user of the package; the same goes for the tests -- they are run on a (virtual version of the) compiled/installed package.

Thus, I have a hard time seeing how this compiler flag would cause these issues during execution.

**Comment by Zachary-Kramer** [comment]
The failures could technically be fixed with the approach in issue #50, but that would only circumvent the issue. I removed this fix recently which is likely what caused these failures, but this shouldn't have caused failures because rSOILWAT2 never reads carbon.in.

Currently, only SOILWAT2 has access to the function that reads carbon.in, so I will have to give rSOILWAT2 access to it. Ultimately, I didn't know that sw_inputDataFromFiles was required to read all input files; I thought the CSV lookup was sufficient.

What is inexplicable, though, is that carbon.in can only be read if RSOILWAT is not defined. There is no other way. There is only one function that touches the file, and it has `#ifndef RSOILWAT` wrapping it. I hope I'm missing something here, because it just doesn't make sense.

> I experience the above errors during run-time as user of the package

Do you experience this only when testing it or in normal runs as well?
